### PR TITLE
X2-1655: authorize net changes x2

### PR DESCRIPTION
**Pull request by sachinp9797**

### CODE REVIEW

File: src/CIMGateway.php [hunk 1]
```diff
@@ -43,6 +43,9 @@ public function getDefaultBillTo()
      */
     public function createCard(array $parameters = array())
     {
+        if (isset($parameters['customerProfileId'])) {
```
Comment (srishaharidas) [inline src/CIMGateway.php]: can you add some tests for these/document what each of these changes are for in the PR description on github

**Comment by sachinp9797** [comment]
**Short description of changes**

- When creating a card using **opaqueDataValue** which is a one time use key, if there is an existing **customerProfileId** for this user then we get **E00039**, which is the error code shown when there is already a duplicate record for this user and we can’t create another customerProfileId.. This happens in **createCard** request, after getting this error the one time use token we had can’t be used further to createPaymentProfile. This is being resolved by creating a **customerPaymentProfile** for the new card.
- The change to **unmask card expiration date** as we need it to get traveler cards in Make a purchase API: /api/users/{id}?private=true. 
- The other change is for generating a **hash based on the opaqueDataValue** in order for each card to have a different hash.

File: src/CIMGateway.php (hunk 1 above)
Comment (sachinp9797) [reply src/CIMGateway.php]: added description, also the same is there in tech doc: https://docs.google.com/document/d/1vqLwqU1IXL6gwuNY78pJ7r_rSOa4NSS-Qv0k3eTacsU/edit#

**Comment by sachinp9797** [comment]
Technical doc for Authorize.net changes: https://docs.google.com/document/d/1vqLwqU1IXL6gwuNY78pJ7r_rSOa4NSS-Qv0k3eTacsU/edit#

**Pull request merged**